color.d

```diff
@@ -128,23 +128,64 @@ struct Color {
 	}
 
 @safe:
-	/++
-		The color components are available as a static array, individual bytes, and a uint inside this union.
 
-		Since it is anonymous, you can use the inner members' names directly.
-	+/
-	union {
-		ubyte[4] components; /// [r, g, b, a]
-
-		/// Holder for rgba individual components.
-		struct {
-			ubyte r; /// red
-			ubyte g; /// green
-			ubyte b; /// blue
-			ubyte a; /// alpha. 255 == opaque
+	ubyte[4] components; /// [r, g, b, a]
+
+	alias components this; /// enables `foreach` etc.
+
+	// The color components are available as individual bytes and a uint through this property functions.
+	// Unlike an anonymous union, this works with CTFE as well.
+	@safe pure nothrow @nogc {
+		// individual rgb components
+		pragma(inline, true) ref inout(ubyte) r() inout scope return { return components[0]; } /// red
```

Comment:
i doubt the pragma is worth it, but since you return `ref` here anyway, no need for a setter; the ref return can be assigned.

As long as this compiles on my oldest supported compiler i think im ok with it, i gotta find where i used these but i don't think anything would break here, it should work fine everywhere and compile to the same thing.

Reply:
I’ve added the setters specifically after something from arsd would no longer compiler

Reply:
I’m not sure how to reproduce this, however

Reply:
I’ve commented out the setters.